rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -461,26 +450,23 @@ private BlockResult executeParallel(
             List<Transaction> sublist = block.getTransactionsList().subList(start, end);
             TransactionListExecutor txListExecutor = new TransactionListExecutor(
                     sublist,
-                    readWrittenKeysTracker,
                     block,
                     transactionExecutorFactory,
-                    track,
+                    track.startTracking(),
                     vmTrace,
                     vmTraceOptions,
-                    deletedAccounts,
+                    new HashSet<>(),
                     discardInvalidTxs,
                     acceptInvalidTransactions,
-                    receipts,
-                    executedTransactions,
-                    transactionResults,
+                    new HashMap<>(),
+                    new HashMap<>(),
+                    new HashMap<>(),
```

Comment:
I would remove these from the constructor as they break encapsulation and can cause problems in the future. Even if it means that we have to manually collect and merge the structures of the last execution. It's just another loop iteration at the end.

Reply:
The last execution needs the merged state. Many of the transactions within the sequential sublist depend on the final state of the parallel execution.